### PR TITLE
Update usage of built-in LLMs in Microsoft Fabric

### PR DESCRIPTION
**Pull request by marklysze**
Update usage of built-in LLMs in Microsoft Fabric, created by @thinkall from forked repo, [PR 3474](https://github.com/microsoft/autogen/pull/3474).

Thanks to @thinkall

## Checks

- [ ] I've included any doc changes needed for https://autogen-ai.github.io/autogen/. See https://autogen-ai.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [X] I've made sure all auto checks have passed.

### CODE REVIEW

File: notebook/agentchat_microsoft_fabric.ipynb [hunk 1]
```diff
@@ -164,7 +404,71 @@
...
+    "import types\n",
+    "\n",
+    "from synapse.ml.fabric.credentials import get_openai_httpx_sync_client\n",
+    "\n",
+    "import autogen\n",
+    "\n",
+    "http_client = get_openai_httpx_sync_client()  # http_client is needed for openai>1\n",
+    "http_client.__deepcopy__ = types.MethodType(\n",
+    "    lambda self, memo: self, http_client\n",
+    ")  # https://microsoft.github.io/autogen/docs/topics/llm_configuration#adding-http-client-in-llm_config-for-proxy\n",
```
Comment (sonichi) [inline notebook/agentchat_microsoft_fabric.ipynb:450]:
```suggestion
    ")  # https://autogen-ai.github.io/autogen/docs/topics/llm_configuration#adding-http-client-in-llm_config-for-proxy\n",
```

File: notebook/agentchat_microsoft_fabric.ipynb [hunk 2]
```diff
@@ -44,31 +42,46 @@
...
      "transient": {
       "deleting": false
      }
     }
    },
+   "outputs": [],
    "source": [
-    "### Set up config_list and llm_config"
+    "# pyautogen<=0.1.14 supports openai<1\n",
+    "%pip install \"pyautogen==0.1.14\" \"openai==0.28.1\" -q"
```
Comment (Hk669) [inline notebook/agentchat_microsoft_fabric.ipynb:84]: are we not using the `autogen` package?

Comment (sonichi) [reply notebook/agentchat_microsoft_fabric.ipynb:84]: Here it's pinning to an older version which only exists for pyautogen.

Comment (marklysze) [reply notebook/agentchat_microsoft_fabric.ipynb:84]: Sorry for the late reply, yes that's correct

**Pull request merged**